apps/api/v1/pages/api/event-types/_post.ts

```diff
@@ -275,26 +275,58 @@ async function postHandler(req: NextApiRequest) {
     children: _,
     ...parsedBody
   } = schemaEventTypeCreateBodyParams.parse(body || {});
-
+  if (parsedBody.teamId) {
+    if (parsedBody.parentId) {
+      // parentId is present only for child event type, teamId is used for event types belonging to team
+      throw new HttpError({
+        statusCode: 400,
+        message: "`parentId` and `teamId` both cannot be present in the request",
+      });
+    }
+    if (parsedBody.userId) {
+      //team event is not associated with an owner or user
+      throw new HttpError({
+        statusCode: 400,
+        message: "`teamId` and `userId` both cannot be present in the request",
+      });
+    }
+  } else {
+    if (parsedBody.schedulingType) {
+      //schedulingType is applicable only for team events
+      throw new HttpError({
+        statusCode: 400,
+        message: "schedulingType is applicable only for team events",
+      });
+    }
+  }
   let data: Prisma.EventTypeCreateArgs["data"] = {
     ...parsedBody,
-    userId: !!parsedBody.teamId ? null : userId,
-    users: !!parsedBody.teamId ? undefined : { connect: { id: userId } },
```

Comment:
This was forcing current user to be connected as user, which was preventing any other user to be added as hosts for managed event type, handled it later.